src/components/views/elements/Quote.js

```diff
@@ -0,0 +1,155 @@
+/*
+Copyright 2017 New Vector Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import React from 'react';
+import sdk from '../../../index';
+import {_t} from '../../../languageHandler';
+import PropTypes from 'prop-types';
+import MatrixClientPeg from '../../../MatrixClientPeg';
+import {MatrixEvent} from 'matrix-js-sdk';
+
+// For URLs of matrix.to links in the timeline which have been reformatted by
+// HttpUtils transformTags to relative links. This excludes event URLs (with `[^\/]*`)
+const REGEX_LOCAL_MATRIXTO = /^#\/room\/(([\#\!])[^\/]*)\/(\$[^\/]*)$/;
+
+const MILLIS_IN_DAY = 86400000;
+function wantsDateSeparator(parentEvent, event) {
+    const parentEventDate = parentEvent.getDate();
```

Comment:
This (as well as REGEX_LOCAL_MATRIXTO) looks duplicated from other places - can we put the function somewhere common?

Reply:
This - yes

`REGEX_LOCAL_MATRIXTO` varies slightly, one is for room only, the other is for room/user
so while yes we could use the latter to capture the same data as the former, it then doesn't offer the check in the same step